django/library/views.py

```diff
@@ -171,14 +173,41 @@ def post(self, request, *args, **kwargs):
         return redirect(request.META.get("HTTP_REFERER", ""))
 
 
-class PeerReviewReviewerListView(mixins.ListModelMixin, viewsets.GenericViewSet):
-    queryset = MemberProfile.objects.all()
-    serializer_class = RelatedMemberProfileSerializer
+class ChangePeerReviewPermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if request.user.has_perm("library.change_peerreview"):
+            return True
+        raise DrfPermissionDenied
+
+
+class PeerReviewerDashboardView(PermissionRequiredMixin, ListView):
+    template_name = "library/review/reviewers.jinja"
+    model = PeerReviewer
+    permission_required = "library.change_peerreview"
+    context_object_name = "reviewers"
+    paginate_by = 15
 
     def get_queryset(self):
-        query = self.request.query_params.get("query", "")
-        results = PeerReview.objects.find_candidate_reviewers(query)
-        return results
+        return PeerReviewer.objects.all()
+
+
+class PeerReviewerFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        # FIXME: when there query, return the full list of reviewers
+        query = request.query_params.get("query", None)
+        if query is None:
+            return queryset
+        return get_search_queryset(query, queryset)
+
+
+class PeerReviewerViewSet(CommonViewSetMixin, NoDeleteViewSet):
+    queryset = PeerReviewer.objects.exclude(is_active=False)
```

Comment:
How would the filtering work here? The main reviews dashboard uses a form but I think that would require using jinja templates.

Reply:
https://www.comses.net/codebases/ has a UI for filtering done in vue (`CodebaseListSidebar.vue`) which just adds on query params to the search/list url which get picked up by the `CodebaseFilter`. Could do the same thing here OR could do the filtering all in client-side JS so its faster

As long as its relatively simple to implement, I think doing it all client-side makes more sense since its a short list and non-paginated

Reply:
the filters I would do for now are:

- include inactive (checkbox)
- name (search box, partial match on name/username)
- language (checkboxes, based on the set of languages available in the whole list)

it might be too messy to try and re-use the existing field components for this, btw